app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -1109,6 +1119,28 @@ BrowserWorld::DistanceToNode(const vrb::NodePtr& aTargetNode, const vrb::Vector&
   return result;
 }
 
+WidgetPtr
+BrowserWorld::GetWidgetFromNode(const vrb::NodePtr& aNode) const {
+  for (const auto & widget: m.widgets) {
+    if (widget->GetRoot() == aNode) {
+      return widget;
+    }
+  }
+  return nullptr;
+}
+
+float
+BrowserWorld::DistanceToPlane(const WidgetPtr& aWidget, const vrb::Vector& aPosition, const vrb::Vector& aDirection) const {
+  if (!aWidget) {
+    return -1.0f;
+  }
+  vrb::Vector result;
+  bool inside;
+  float distance;
```

Comment:
nit: I would prefer `inside` and `distance` have default values.

Reply:
done